Mathlib/Analysis/Normed/Group/SemiNormedGroup.lean

```diff
@@ -0,0 +1,297 @@
+
+We define `SemiNormedGroup`, the category of seminormed groups and normed group homs between them,
+as well as `SemiNormedGroup₁`, the subcategory of norm non-increasing morphisms.
+-/
+
+
+noncomputable section
+
+universe u
+
+open CategoryTheory
+
+/-- The category of seminormed abelian groups and bounded group homomorphisms. -/
+def SemiNormedGroup : Type (u + 1) :=
+  Bundled SeminormedAddCommGroup
+
+set_option linter.uppercaseLean3 false --in [linter.uppercaseLean3]
+#align SemiNormedGroup SemiNormedGroup
+
+namespace SemiNormedGroup
+
+instance bundledHom : BundledHom @NormedAddGroupHom where
+  toFun := @NormedAddGroupHom.toFun
+  id := @NormedAddGroupHom.id
+  comp := @NormedAddGroupHom.comp
+
+#align SemiNormedGroup.bundled_hom SemiNormedGroup.bundledHom
+
+deriving instance LargeCategory for SemiNormedGroup
+
+--Porting note: deriving fails for ConcreteCategory, adding instance manually.
+--deriving instance LargeCategory, ConcreteCategory for SemiRingCat
+instance : ConcreteCategory SemiNormedGroup := by
+  dsimp [SemiNormedGroup]
+  infer_instance
+
+instance : CoeSort SemiNormedGroup (Type _) where
+  coe X := X.α
+
+/-- Construct a bundled `SemiNormedGroup` from the underlying type and typeclass. -/
+def of (M : Type u) [SeminormedAddCommGroup M] : SemiNormedGroup :=
+  Bundled.of M
+#align SemiNormedGroup.of SemiNormedGroup.of
+
+instance (M : SemiNormedGroup) : SeminormedAddCommGroup M :=
+  M.str
+
+/-- Porting Note: Added -- needed to make coe_id work-/
+instance {X Y : SemiNormedGroup} : CoeFun (X ⟶ Y) fun _ => X → Y where
+  coe (f : X ⟶ Y) := NormedAddGroupHom.toFun f
+
+@[simp]
+theorem coe_of (V : Type u) [SeminormedAddCommGroup V] : (SemiNormedGroup.of V : Type u) = V :=
+  rfl
+#align SemiNormedGroup.coe_of SemiNormedGroup.coe_of
+
+-- Porting note : marked with high priority to short circuit simplifier's path
+@[simp (high)]
+theorem coe_id (V : SemiNormedGroup) : (𝟙 V : V → V) = id :=
+  rfl
+#align SemiNormedGroup.coe_id SemiNormedGroup.coe_id
+
+-- Porting note : marked with high priority to short circuit simplifier's path
+@[simp (high)]
+theorem coe_comp {M N K : SemiNormedGroup} (f : M ⟶ N) (g : N ⟶ K) : (f ≫ g : M → K) = g ∘ f :=
+  rfl
+#align SemiNormedGroup.coe_comp SemiNormedGroup.coe_comp
+
+instance : Inhabited SemiNormedGroup :=
+  ⟨of PUnit⟩
+
+instance ofUnique (V : Type u) [SeminormedAddCommGroup V] [i : Unique V] :
+    Unique (SemiNormedGroup.of V) :=
+  i
+#align SemiNormedGroup.of_unique SemiNormedGroup.ofUnique
+
+/--porting note: originally empty, which didn't work. Notation for composition changed?-/
+instance : Limits.HasZeroMorphisms.{u, u + 1} SemiNormedGroup where
+  Zero _ _ := NormedAddGroupHom.zero
+  comp_zero _ _ := rfl
+  zero_comp _ _ _ f := NormedAddGroupHom.comp_zero f
+
+
+@[simp]
+theorem zero_apply {V W : SemiNormedGroup} (x : V) : (0 : V ⟶ W) x = 0 :=
+  rfl
+#align SemiNormedGroup.zero_apply SemiNormedGroup.zero_apply
+
+/--Porting note: Had to add some proof steps -/
```

Comment:
I won't insist that it happen here, but it is *super* helpful if you can include the original proof in a comment. It makes it much easier if someone wants to try to diagnose it later (or indeed just wants to classify what sort of failure it was).

Reply:
As in, why did you have to add some steps? Was it a problem with `simp`? unification? typeclass search?

Reply:
Thanks! I tried to add more detail.